src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -3324,14 +3326,33 @@ bool BlockchainLMDB::get_blocks_from(uint64_t start_height, size_t min_block_cou
         throw0(DB_ERROR(lmdb_error("Error attempting to retrieve transaction data from the db: ", result).c_str()));
       tx_blob.assign((const char*)v.mv_data, v.mv_size);
 
-      if (!pruned)
-      {
+      if (pruned) {
+        // get the prunable hash
+
+        // Look up each transaction's tx_id
```

Comment:
We can skip the TXID -> tx index lookup by using already existing `tx_id` (terrible name 😢) variable. Tx on-chain indices are assigned in ascending order by 1) block, then 2) txs in block in specified order.